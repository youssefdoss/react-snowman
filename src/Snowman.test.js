import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Snowman from './Snowman';

let game;

const img0 = 'img0';
const img1 = 'img1';
const img2 = 'img2';

beforeEach(function() {
  game = render(<Snowman
    words={['apple']}
    maxWrong={3}
    images={[img0, img1, img2]}/>);
});

/** clickLetter: clicks on a letter in an effort to aid tests
 *
 * letter: string
 */

function clickLetter(letter) {
  fireEvent.click(game.getByText(letter));
}

// Feels redundant given beforeEach
it('renders successfully', function() {
  render(<Snowman />);
});

it('matches initial snapshot', function() {
  expect(game).toMatchSnapshot();
});

it('matches snapshot after loss', function() {
  clickLetter('b');
  clickLetter('c');
  clickLetter('d');

  expect(game).toMatchSnapshot();
});

it('only allows maxWrong guesses and renders correct endgame', function() {
  clickLetter('b');
  clickLetter('c');
  clickLetter('d');

  expect(game.getByText('You lose')).toBeInTheDocument();
});