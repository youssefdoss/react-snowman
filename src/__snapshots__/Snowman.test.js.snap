// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches initial snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Snowman"
      >
        <img
          alt="0"
          src="img0"
        />
        <p>
          Number wrong: 
          0
        </p>
        <p
          class="Snowman-word"
        >
          _
          _
          _
          _
          _
        </p>
        <p>
          <button
            value="a"
          >
            a
          </button>
          <button
            value="b"
          >
            b
          </button>
          <button
            value="c"
          >
            c
          </button>
          <button
            value="d"
          >
            d
          </button>
          <button
            value="e"
          >
            e
          </button>
          <button
            value="f"
          >
            f
          </button>
          <button
            value="g"
          >
            g
          </button>
          <button
            value="h"
          >
            h
          </button>
          <button
            value="i"
          >
            i
          </button>
          <button
            value="j"
          >
            j
          </button>
          <button
            value="k"
          >
            k
          </button>
          <button
            value="l"
          >
            l
          </button>
          <button
            value="m"
          >
            m
          </button>
          <button
            value="n"
          >
            n
          </button>
          <button
            value="o"
          >
            o
          </button>
          <button
            value="p"
          >
            p
          </button>
          <button
            value="q"
          >
            q
          </button>
          <button
            value="r"
          >
            r
          </button>
          <button
            value="s"
          >
            s
          </button>
          <button
            value="t"
          >
            t
          </button>
          <button
            value="u"
          >
            u
          </button>
          <button
            value="v"
          >
            v
          </button>
          <button
            value="w"
          >
            w
          </button>
          <button
            value="x"
          >
            x
          </button>
          <button
            value="y"
          >
            y
          </button>
          <button
            value="z"
          >
            z
          </button>
        </p>
        <button>
          Restart
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Snowman"
    >
      <img
        alt="0"
        src="img0"
      />
      <p>
        Number wrong: 
        0
      </p>
      <p
        class="Snowman-word"
      >
        _
        _
        _
        _
        _
      </p>
      <p>
        <button
          value="a"
        >
          a
        </button>
        <button
          value="b"
        >
          b
        </button>
        <button
          value="c"
        >
          c
        </button>
        <button
          value="d"
        >
          d
        </button>
        <button
          value="e"
        >
          e
        </button>
        <button
          value="f"
        >
          f
        </button>
        <button
          value="g"
        >
          g
        </button>
        <button
          value="h"
        >
          h
        </button>
        <button
          value="i"
        >
          i
        </button>
        <button
          value="j"
        >
          j
        </button>
        <button
          value="k"
        >
          k
        </button>
        <button
          value="l"
        >
          l
        </button>
        <button
          value="m"
        >
          m
        </button>
        <button
          value="n"
        >
          n
        </button>
        <button
          value="o"
        >
          o
        </button>
        <button
          value="p"
        >
          p
        </button>
        <button
          value="q"
        >
          q
        </button>
        <button
          value="r"
        >
          r
        </button>
        <button
          value="s"
        >
          s
        </button>
        <button
          value="t"
        >
          t
        </button>
        <button
          value="u"
        >
          u
        </button>
        <button
          value="v"
        >
          v
        </button>
        <button
          value="w"
        >
          w
        </button>
        <button
          value="x"
        >
          x
        </button>
        <button
          value="y"
        >
          y
        </button>
        <button
          value="z"
        >
          z
        </button>
      </p>
      <button>
        Restart
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches snapshot after loss 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Snowman"
      >
        <img
          alt="3"
        />
        <p>
          Number wrong: 
          3
        </p>
        <p
          class="Snowman-word"
        >
          _
          _
          _
          _
          _
        </p>
        <p>
          You lose
        </p>
        <button>
          Restart
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Snowman"
    >
      <img
        alt="3"
      />
      <p>
        Number wrong: 
        3
      </p>
      <p
        class="Snowman-word"
      >
        _
        _
        _
        _
        _
      </p>
      <p>
        You lose
      </p>
      <button>
        Restart
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
